### PR TITLE
Audit Trail Support

### PR DESCRIPTION
**Pull request by henrylamchan**
# Overview
[Asana](https://app.asana.com/0/1171029126931209/1170059789104317/f)
Add Audit Trail support.. plus some things.

Notable changes include:
- Support for Audit Trail
- Bit of refactoring on how users of client say they want to include an auth header
- Posting as JSON
- Storing the raw response in profile... just in case
- Rip out the validation stuff and place checks in WorkOS.php instead

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,12 +11,7 @@ composer require workos/workos-php
 
 ## Getting Started
 
-The package will need to be configured with your [api key](https://dashboard.workos.com/api-keys) and [project id](https://dashboard.workos.com/sso/configuration):
-
-```php
-\WorkOS\WorkOS::setApiKey('sk_123secret456');
-\WorkOS\WorkOS::setProjectId('project_456demo789');
-```
+The package will need to be configured with your [api key](https://dashboard.workos.com/api-keys) and [project id](https://dashboard.workos.com/sso/configuration). By default, the packages looks for a `WORKOS_API_KEY` and `WORKOS_PROJECT_ID` environment variable.
```
Comment (rjvani) [inline README.md:14]: Nit: api key -> API Key

Comment (rjvani) [reply README.md:14]: Nit: project id -> Project ID

File: README.md [hunk 2]
```diff
@@ -38,3 +33,26 @@ the code passed back from WorkOS to grab the profile of the authenticated user t
 ```php
 $profile = (new \WorkOS\SSO())->getProfile($code);
 ```
+
+### Audit Trail
+Creating an Audit Trail event requires a descriptive action name and annotating the event with its CRUD identifier. The action name must contain an action category and an action name separated by a period, for example, user.login.
```
Comment (rjvani) [inline README.md:38]: Nit: Can you wrap user.login with ticks for Markdown rendering: `user.login`?

Comment (henrylamchan) [reply README.md:38]: Actually, this whole bit was leftover copy pasta from the docs that I wanted to get rid of. Gonna leave a quick example of audit trail here and if users want to know more, they can check out our docs site.

**Pull request merged**